java/client/src/org/openqa/selenium/remote/ErrorHandler.java

```diff
@@ -220,7 +220,7 @@ private String duration(long duration) {
 
   private Throwable rebuildServerError(Map<String, Object> rawErrorData, int responseStatus) {
 
-    if (!rawErrorData.containsKey(CLASS) && !rawErrorData.containsKey(STACK_TRACE)) {
+    if (rawErrorData.get(CLASS) == null || rawErrorData.get(STACK_TRACE) == null) {
```

Comment:
Why && changed to || ?

Reply:
The check makes sure that valid values are present for CLASS and STACK_TRACE. So we want to make sure that both CLASS is not null and STACK_TRACE is not null. In other words, quit if CLASS or STACK_TRACE are null.

Reply:
Ah, I missed that.

I don't think that comment holds true in the W3C WebDriver era - see https://www.w3.org/TR/webdriver/#handling-errors.

`class` doesn't seem to be defined in the spec and `stacktrace` is "implementation-specific".

What this code tries to do, is to reconstruct a Java exception based on the error data sent by the server. But there is no guarantee at all that the server is implemented in Java.

Reply:
Java binding tries to be backward compatible. At least until we turn to 4.x

Reply:
That's fair, but in this case it would cause the binding to crash when a remote end sends W3C-spec compliant responses 😄 .

Reply:
So we should try to make it work for both legacy and W3C conformant cases.